src/target/adiv5.c

```diff
@@ -294,9 +295,10 @@ static uint32_t adiv5_mem_read32(ADIv5_AP_t *ap, uint32_t addr)
 static uint32_t adiv5_ap_read_id(ADIv5_AP_t *ap, uint32_t addr)
 {
 	uint32_t res = 0;
+	uint8_t data[16];
+	adiv5_mem_read(ap, data, addr, sizeof(data));
 	for (int i = 0; i < 4; i++) {
-		uint32_t x = adiv5_mem_read32(ap, addr + 4 * i);
-		res |= (x & 0xff) << (i * 8);
+		res |= (data[4 * i] << (i * 8));
```

Comment:
The extra ()'s around the outside of the operation are redundant and suppress warnings, please leave them out - ie `res |= data[i * 4U] << (i * 8U);`